img/src/ChargeSolving.cxx

```diff
@@ -85,15 +86,69 @@ void blob_weight_simple(const cluster_graph_t& cgraph, graph_t& csg)
 }
 // fixme: implement distance.
 
+
+void blob_weight_uboone(const cluster_graph_t& cgraph, graph_t& csg)
+{
+    int nblobs = 0;
+    for (auto desc : vertex_range(csg)) {
+        auto& vtx = csg[desc];
+        if (vtx.kind != node_t::blob) {
+            continue;
+        }
+        ++nblobs;
+        // check if blob is connected to other blobs in other slices
+        // connection | weight
+        // non | 9
+        // one | 3
+        // both| 1
+        auto cent_time = (int)csg[boost::graph_bundle].islice->start();
+        bool prev_con = false;
+        bool next_con = false;
+        for (auto edge : mir(boost::out_edges(vtx.orig_desc, cgraph)))
+        {
+            vdesc_t ndesc = boost::target(edge, cgraph);
+            const auto& nnode = cgraph[ndesc];
+            if (nnode.code() == 'b') {
+                auto time = (int)get<blob_t>(nnode.ptr)->slice()->start();
+                if (time > cent_time) {
+                    next_con = true;
+                }
+                if (time < cent_time) {
+                    prev_con = true;
+                }
+            }
+        }
+        double weight = 9.;
+        if (next_con) {
+            weight /= 3.;
+        }
+        if (prev_con) {
+            weight /= 3.;
+        }
+        vtx.value.uncertainty((float) weight);
+        // TODO remove this
+        // std::cout << String::format("cent_time: %f next_con: %d, prev_con: %d, weight: %d", cent_time, next_con, prev_con, weight) << std::endl;
+    }
+    // TODO remove this
+    // std::cout << String::format("nblobs: %d", nblobs) << std::endl;
+}
+
 // Weighting function lookup
 using blob_weighting_f = std::function<void(const cluster_graph_t& cgraph, graph_t& csg)>;
 using blob_weighting_lut = std::unordered_map<std::string, blob_weighting_f>;
 static blob_weighting_lut gStrategies{
     {"uniform", blob_weight_uniform},
     {"simple", blob_weight_simple},
+    {"uboone", blob_weight_uboone},
     // distance....
 };
 
+
+static std::unordered_map<std::string, SolveParams::Config> gSolveParamsConfigMap{
+    {"simple", SolveParams::simple},
+    {"uboone", SolveParams::uboone},
+};
+
```

Comment:
These globals are not so great.  At the very least they should be made `const`.  But better would be to resolve the policy/strategy to a type erased object held by the instance.  Eg, as a `std::function` type.  The policy resolution would be done in `configure()`.